### PR TITLE
added tag queries to db, test script in useeffect in user-groups page

### PR DESCRIPTION
**Pull request by dionyichia**
### 🦋 Description
[//]: # "Required - Describe what's new in this PR in a few lines. A description and bullet points for specifics will suffice."

Added queries for tag, 

### 🦧 Screenshots
[//]: # "Required for frontend changes, otherwise optional but strongly recommended. Add screenshots of expected behavior - GIFs if you're feeling fancy! Use the provided image template. Drag the desired image into the PR, then copy the link into the placeholder."

[image placeholder]: 

image

## 🐸 How to review
[//]: # 'Required - Describe the order in which to review files and what to expect when testing locally. Is there anything specifically you want feedback on? Should this be reviewed commit by commit, or all at once? What are some user flows to test? What are some edge cases to look out for?'

 got to /user-groups/ page, check console logs

## 🐙 Next steps
[//]: # "Optional - What's NOT in this PR, doesn't work yet, and/or still needs to be done. Note any temporary fixes in this PR that should be cleaned up later."

remove use effect to integrate

### Online sources
[//]: # 'Copy links to any tutorials or documentation that was useful to you when working on this PR'

### 🐁 Related PRs
[//]: # "Add related PRs you're waiting on/ PRs that will conflict, etc; if this is a refactor, feel free to add PRs that previously modified this code"

CC: @me-liu

### CODE REVIEW

File: actions/supabase/queries/tag.ts [hunk 1]
```diff
@@ -0,0 +1,117 @@
...
+  if (error) {
+    throw new Error(`Error fetching tags for template: ${error.message}`);
+  }
+
+  if (!data) {
+    return [];
+  }
+
+  // Extract the tag objects from the wrapper
+  return data.map((item: Dict) => item.tag);
```
Comment (me-liu) [inline actions/supabase/queries/tag.ts]: I think the Dict here may be too loosely typed. Can you make something like
```
type templateTag = { tag: Tag };
type tagTemplate = { template: Template };
```
and then use those as dicts instead of Dict from styledcomponents?

**Review by me-liu — COMMENTED** [review]
solid work dion

File: actions/supabase/queries/tag.ts (hunk 1 above)
Comment (dionyichia) [reply actions/supabase/queries/tag.ts]: okay dones

**Review by me-liu — APPROVED** [review]
